pkg/bootstrap/render.go

```diff
@@ -171,10 +172,14 @@ func (b *KlusterletManifestsConfig) Generate(ctx context.Context, clientHolder *
 	var kcRegistries []klusterletconfigv1alpha1.Registries
 	var kcNodePlacement *operatorv1.NodePlacement
 	var kcImagePullSecret corev1.ObjectReference
+	var appliedManifestWorkEvictionGracePeriod string
 	if b.klusterletconfig != nil {
-		kcRegistries = b.klusterletconfig.Spec.Registries
-		kcNodePlacement = b.klusterletconfig.Spec.NodePlacement
-		kcImagePullSecret = b.klusterletconfig.Spec.PullSecret
+		if b.InstallMode == operatorv1.InstallModeDefault || b.InstallMode == operatorv1.InstallModeSingleton {
```

Comment:
hosted mode can not be configured?

Reply:
It's not quite clear where the configuration of registries, node placement and pull secret should come from, the klusterletconfig of the managed cluster or klusterletconfig of the hosting cluster? It will be tracked as a bug (https://issues.redhat.com/browse/ACM-11532) later.